src/platform/synchronization.rs

```diff
@@ -0,0 +1,291 @@
+use std::{
+    sync::{Condvar, Mutex, MutexGuard},
+    thread::{self, JoinHandle},
+};
+
+use libc::c_void;
+
+use crate::{impl_guarded_transmute, GuardedTransmute};
+
+pub struct ZMutex<'a> {
+    mutex: Mutex<()>,
+    lock: Option<MutexGuard<'a, ()>>,
+}
+
+pub struct ZMutexPtr {
+    data: Option<Box<ZMutex<'static>>>,
+}
+
+/// Mutex
+///
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub struct z_mutex_t(usize);
+
+impl_guarded_transmute!(z_mutex_t, ZMutexPtr);
+impl_guarded_transmute!(ZMutexPtr, z_mutex_t);
+
+const EBUSY: i8 = -1;
```

Comment:
As far as I see these error values doubles the ones from pthread library (https://pubs.opengroup.org/onlinepubs/7908799/xsh/pthread_mutex_init.html). This should be at least commented and maybe exposed in the documentation.